src/Microsoft.DotNet.Darc/Darc/Operations/GetChannelOperation.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Darc.Options;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Maestro.Client;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Darc.Operations;
+
+internal class GetChannelOperation : Operation
+{
+    private readonly GetChannelCommandLineOptions _options;
+    public GetChannelOperation(GetChannelCommandLineOptions options)
+        : base(options)
+    {
+        _options = options;
+    }
+
+    /// <summary>
+    /// Retrieve information about a specific channel
+    /// </summary>
+    /// <param name="options">Command line options</param>
+    /// <returns>Process exit code.</returns>
+    public override async Task<int> ExecuteAsync()
+    {
+        try
+        {
+            IBarApiClient barClient = Provider.GetRequiredService<IBarApiClient>();
+
+            var channel = await barClient.GetChannelAsync(_options.Id);
+
+            if (channel == null)
+            {
+                Logger.LogError("Channel with id {channelId} not found", _options.Id);
+                return Constants.ErrorCode;
```

Comment:
Shouldn't the error code for not found be a different one from others to distinguishing the case?

Reply:
A fair point but unfortunately, this old tool does not have other codes than success and error and we use those well known exit codes everywhere